### PR TITLE
assignment_rspec_calculator

### PR DESCRIPTION
**Pull request by chadl76**

### CODE REVIEW

File: spec/calculator_spec.rb [hunk 1]
```diff
@@ -1 +1,72 @@
-# Your code here
\ No newline at end of file
+# Your code here
+require 'calculator.rb'
+
+describe Calculator do
+
```
Comment (remis1889) [inline spec/calculator_spec.rb:5]: create a common instance for `Calculator` and use it throughout the specs.
`let(:c) { Calculator.new }`

File: spec/calculator_spec.rb [hunk 2]
```diff
@@ -1 +1,72 @@
...
+	describe '#add' do
+		it 'returns the sum of its arguments' do
+      expect(Calculator.new.add(1,2)).to eq(3)
+    end
+
+    it 'properly adds two negative numbers' do 
+    	expect(Calculator.new.add(-1,-2)).to eq(-3)
+    end
```
Comment (remis1889) [inline spec/calculator_spec.rb:13]: test for floating point numbers too.

File: spec/calculator_spec.rb [hunk 3]
```diff
@@ -1 +1,72 @@
...
+
+  describe '#divide' do
+  	it 'should raise error if dividing by 0' do
+  		expect{Calculator.new.divide(4,0)}.to raise_error(ArgumentError)
+  	end
+
+  	it 'should divide two integers without remainder' do
+  		expect(Calculator.new.divide(8,2)).to eq(4)
+  	end
+  end
```
Comment (remis1889) [inline spec/calculator_spec.rb:48]: You should also test if the result of a division with remainder is a `Float`

File: spec/calculator_spec.rb [hunk 4]
```diff
@@ -1 +1,72 @@
...
+
+  describe '#pow' do
+  	it 'should raise number to its power' do
+  		expect(Calculator.new.pow(2,3)).to eq(8)
+  	end
+    it 'should raise number to its negative power' do
+    	expect(Calculator.new.pow(2,-3)).to eq(2 ** -3)
+    end
```
Comment (remis1889) [inline spec/calculator_spec.rb:56]: Also, test for cases when the exponent is zero and a decimal

File: spec/calculator_spec.rb [hunk 5]
```diff
@@ -1 +1,72 @@
...
+  end
+
+  describe '#sqrt' do
+  	it 'should return the square root of integer' do
+  		expect(Calculator.new.sqrt(16)).to eq(Math.sqrt(16))
```
Comment (remis1889) [inline spec/calculator_spec.rb:61]: also, write tests for negative argument, round roots and non-round roots.

File: spec/calculator_spec.rb [hunk 6]
```diff
@@ -1 +1,72 @@
...
+  	end
+  end
+
+  describe '#stringify' do
+  	it 'should return answer as a string' do
+      expect(Calculator.new(true).add(1,1)).to eq(2.to_s)
```
Comment (remis1889) [inline spec/calculator_spec.rb:67]: write tests for other main operations too when `stringify` is `true`

**Review by remis1889 — COMMENTED** [review]
Missing tests for `memory` and `memory=` methods
